server/sonar-server/src/main/java/org/sonar/server/user/ws/DeactivateAction.java

```diff
@@ -83,21 +80,21 @@ public void handle(Request request, Response response) throws Exception {
   }
 
   private void writeResponse(Response response, String login) {
-    UserDoc user = index.getByLogin(login);
     JsonWriter json = response.newJsonWriter().beginObject();
-    writeUser(json, user);
+    writeUser(json, login);
     json.endObject().close();
   }
 
-  private void writeUser(JsonWriter json, UserDoc user) {
+  private void writeUser(JsonWriter json, String login) {
     json.name("user");
     Set<String> groups = Sets.newHashSet();
     DbSession dbSession = dbClient.openSession(false);
     try {
-      groups.addAll(dbClient.groupMembershipDao().selectGroupsByLogins(dbSession, Arrays.asList(user.login())).get(user.login()));
+      UserDto user = dbClient.userDao().selectOrFailByLogin(dbSession, login);
```

Comment:
here `dbClient.userDao().selectOrFailByLogin(dbSession, login)` shouldn't be used. The dao layer returns a 500 HTTP error whereas we want a 401 in this specific case, as it's a user input. Si ComponentFinder for example, that return server exception